packages/system/src/Box.tsx

```diff
@@ -69,8 +69,19 @@ export type BoxProps = PolymorphicPropsWithRef<BoxOwnProps, 'div'>;
 const isNotEmpty = (val: any) =>
   !(val && Object.keys(val).length === 0 && val.constructor === Object);
 
-const ensureArray = <T extends any>(val?: T | T[]) =>
-  Array.isArray(val) ? val : [val];
+/**
+ * Ensure that variant is an array and remove dot at the end of the string
+ * to set __default in theme without giving a default variant to the component
+ */
+const ensureCorrectVariant = <T extends string>(variant?: T | T[]) => {
```

Comment:
Can we make this function more simple to read please? :D

Reply:
Better? Or how can I make it easier?